cli/src/config.rs

```diff
@@ -215,58 +246,62 @@ impl UnresolvedConfigEnv {
                 .map(ConfigPath::new)
                 .collect();
         }
-        let mut paths = vec![];
-        let home_config_path = self.home_dir.map(|mut home_dir| {
-            home_dir.push(".jjconfig.toml");
-            ConfigPath::new(home_dir)
-        });
-        let platform_config_path = self.config_dir.clone().map(|mut config_dir| {
-            config_dir.push("jj");
-            config_dir.push("config.toml");
-            ConfigPath::new(config_dir)
-        });
-        let platform_config_dir = self.config_dir.map(|mut config_dir| {
-            config_dir.push("jj");
-            config_dir.push("conf.d");
-            ConfigPath::new(config_dir)
-        });
 
-        use ConfigPath::*;
+        let push_if_exists = |v: &mut Vec<ConfigPath>, path: ConfigPath| {
+            if path.exists() {
+                v.push(path);
+            }
+        };
 
-        match home_config_path {
-            Some(path @ Existing(_)) => paths.push(path),
-            Some(path @ New(_)) if platform_config_path.is_none() => paths.push(path),
-            _ => (),
+        let mut paths = vec![];
+        if let Some(mut home_dir) = self.home_dir {
+            home_dir.push(".jjconfig.toml");
+            let home_dir_config = ConfigPath::new(home_dir);
+            if self.config_dir.is_none() {
+                paths.push(home_dir_config);
+            } else {
+                push_if_exists(&mut paths, home_dir_config);
+            }
         }
 
         if cfg!(target_os = "macos") {
-            let legacy_platform_config_path =
-                self.macos_legacy_config_dir.clone().map(|mut config_dir| {
-                    config_dir.push("jj");
-                    config_dir.push("config.toml");
-                    ConfigPath::new(config_dir)
-                });
-            let legacy_platform_config_dir = self.macos_legacy_config_dir.map(|mut config_dir| {
-                config_dir.push("jj");
-                config_dir.push("conf.d");
-                ConfigPath::new(config_dir)
-            });
-
-            if let Some(path @ Existing(_)) = legacy_platform_config_path {
-                paths.push(path);
-            }
-            if let Some(path @ Existing(_)) = legacy_platform_config_dir {
-                paths.push(path);
+            if let Some(mut legacy_platform_config_path) = self.macos_legacy_config_dir {
```

Comment:
nit: I suppose we can drop the `if cfg!(target_os = "macos")` condition now since `self. macos_legacy_config_dir` would be `None` on other platforms anyway